compiler/rustc_codegen_rmc/src/compiler_interface.rs

```diff
@@ -126,12 +129,22 @@ impl CodegenBackend for GotocCodegenBackend {
             &tcx.sess.opts.debugging_opts.symbol_table_passes,
         );
 
-        let type_map = BTreeMap::from_iter(c.type_map.into_iter().map(|(k, v)| (k, v.to_string())));
+        // Map MIR types to GotoC types
+        let type_map =
+            BTreeMap::from_iter(c.type_map.into_iter().map(|(k, v)| (k, v.to_string().into())));
```

Comment:
Why do we need to do this, instead of just keeping the `<String, type>` map?

Reply:
Presumably because `TyS` doesn't have a serialization, the only thing I'm touching here is changing the string to be interned so I would rather not change more.

Reply:
OK